api/v1alpha2/awsmachine_types.go

```diff
@@ -88,6 +88,11 @@ type AWSMachineSpec struct {
 	// +optional
 	// +kubebuilder:validation:MaxItems=2
 	NetworkInterfaces []string `json:"networkInterfaces,omitempty"`
+
+	// EncryptedCloudInit will use SSM Parameter store to encrypt cloud-init based
+	// userdata
+	// +optional
+	EncryptedCloudInit *bool `json:"encryptedCloudInit,omitempty"`
```

Comment:
Alternatively, since this is CAPA specific, possibly `EncryptUserData`, which is more indicative of what it does.

Reply:
Any particular reason to add this to the v1alpha2 types rather than backup/restore the field like we are doing in CAPI? It would avoid potential data loss from a client that doesn't have the updated v1alpha2 type with the field added.

Reply:
This is probably going to be backported in some way. Not sure about ordering. In any case, removed the v1alpha2 types for now.